### ISSUE TITLE
grgit dependendency should be pinned to a specific version to avoid breaking changes

### ISSUE THREAD

**Issue by ciscoo** [ajoberstar/gradle-git-publish#97]
Coming into work this morning, a library that utilizes this plugin started failing with the following:

```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':documentation'.
> Could not resolve all files for configuration ':documentation:classpath'.
   > Could not resolve org.ajoberstar.grgit:grgit-core:latest.release.
     Required by:
         project :documentation > org.ajoberstar.git-publish:org.ajoberstar.git-publish.gradle.plugin:3.0.0 > org.ajoberstar:gradle-git-publish:3.0.0
      > No matching variant of org.ajoberstar.grgit:grgit-core:5.0.0-rc.1 was found. The consumer was configured to find a runtime of a library compatible with Java 8, packaged as a jar, and its dependencies declared externally, as well as attribute 'org.gradle.plugin.api-version' with value '7.4-rc-1' but:
          - Variant 'apiElements' capability org.ajoberstar.grgit:grgit-core:5.0.0-rc.1 declares a library, packaged as a jar, and its dependencies declared externally:
              - Incompatible because this component declares an API of a component compatible with Java 11 and the consumer needed a runtime of a component compatible with Java 8
              - Other compatible attribute:
                  - Doesn't say anything about org.gradle.plugin.api-version (required '7.4-rc-1')
          - Variant 'javadocElements' capability org.ajoberstar.grgit:grgit-core:5.0.0-rc.1 declares a runtime of a component, and its dependencies declared externally:
              - Incompatible because this component declares documentation and the consumer needed a library
              - Other compatible attributes:
                  - Doesn't say anything about its target Java version (required compatibility with Java 8)
                  - Doesn't say anything about its elements (required them packaged as a jar)
                  - Doesn't say anything about org.gradle.plugin.api-version (required '7.4-rc-1')
          - Variant 'runtimeElements' capability org.ajoberstar.grgit:grgit-core:5.0.0-rc.1 declares a runtime of a library, packaged as a jar, and its dependencies declared externally:
              - Incompatible because this component declares a component compatible with Java 11 and the consumer needed a component compatible with Java 8
              - Other compatible attribute:
                  - Doesn't say anything about org.gradle.plugin.api-version (required '7.4-rc-1')
          - Variant 'sourcesElements' capability org.ajoberstar.grgit:grgit-core:5.0.0-rc.1 declares a runtime of a component, and its dependencies declared externally:
              - Incompatible because this component declares documentation and the consumer needed a library
              - Other compatible attributes:
                  - Doesn't say anything about its target Java version (required compatibility with Java 8)
                  - Doesn't say anything about its elements (required them packaged as a jar)
                  - Doesn't say anything about org.gradle.plugin.api-version (required '7.4-rc-1')

* Try:
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Exception is:
org.gradle.api.ProjectConfigurationException: A problem occurred configuring project ':documentation'.
```

What stood out to me immediately was `org.ajoberstar.grgit:grgit-core:latest.release`. I was not aware that this plugin always pulls in the latest version.

Taking a look at commits for `grgit-core`, I see the following: https://github.com/ajoberstar/grgit/commit/1449adab7907b21e19f9c6bc4c216130ffb00dbe

I suspect this is the main issue that is causing the build to fail. In any case, `gradle-git-publish` should used specific versions rather than always using the latest to avoid pulling in any breaking changes.

**Comment by ciscoo** [comment]
Fixed by forcing a specific version:

```kotlin
buildscript {
    dependencies {
        classpath("org.ajoberstar.grgit:grgit-core") {
            version {
                strictly("4.1.1")
                because("grgit-core uses 'latest.release' and later releases drop Java 8 support due to JGit 6.")
            }
        }
    }
}

plugins {
    id("org.ajoberstar.git-publish") version "3.0.0"
}
```

**Comment by ajoberstar** [comment]
Releasing fix as 3.1.1. This is the same issue I had with reckon, so copying my explanation over:

This should be fixed now in 3.1.1. It was caused by some interplay with com.gradle.plugin-publish, java-gradle-plugin, and dependency locking. I believe I've got a solution that's more reliable now (which includes not publishing directly to the Gradle plugin portal anymore, and making use of the fact that the portal proxies Maven Central).

It may take up to an hour for the version to show up in Central (due to their syncing process), but it should resolve the issue you all encountered.
